File: php-pr/51/php/code-quality-error-handling/JsonParser.php
<?php

class JsonParser
{
    public function parseJson($jsonString)
    {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        try {
            $data = json_decode($jsonString, true);
            return $data;
        } catch (Exception $e) {
            throw new Exception("Something went wrong"); // Generic exception
        }
        // {/fact}
    }
}

File: ruby-pr/53/ruby/code-quality-error-handling/json_parser.rb
require 'json'

class JsonParser
  def parse_json(json_string)
    # {fact rule=code-quality-error-handling@v1.0 defects=1}
    begin
      data = JSON.parse(json_string)
      return data
    rescue => e
      raise StandardError, "Something went wrong" # Generic exception
    end
    # {/fact}
  end
end